src/test/java/GroupDAOTest.java

```diff
@@ -0,0 +1,138 @@
+import org.junit.Test;
+import org.junit.*;
+import org.junit.rules.ExpectedException;
+import server.database.dao.GroupDAO;
+
+import java.io.*;
+
+public class GroupDAOTest {
+    private static String testGroups = "./src/test/data/testGroups.json";
+    private static OutputStream outputStream;
+
+    @BeforeClass
+    @AfterClass
+    public static void clearFile() throws IOException {
+        outputStream = new FileOutputStream(testGroups);
+        outputStream.write("".getBytes());
+        outputStream.close();
+    }
+
+    @Before
+    public void getFile() throws IOException {
+       outputStream = new FileOutputStream(testGroups);
+    }
+
+    @After
+    public void closeFile() throws IOException {
+       outputStream.close();
+    }
+
+    @Rule
+    public ExpectedException expectedEx = ExpectedException.none();
+
+    @Test
+    public void valid() throws InterruptedException, IOException {
+        // Arrange
+        String content =
+                "[\n" +
+                "  {\n" +
+                "    \"groupID\": \"1\",\n" +
+                "    \"groupName\": \"top\",\n" +
+                "    \"parent\": null\n" +
+                "  }, {\n" +
+                "    \"groupID\": \"2\",\n" +
+                "    \"groupName\": \"middle\",\n" +
+                "    \"parent\": \"1\"\n" +
+                "  }, {\n" +
+                "    \"groupID\": \"3\",\n" +
+                "    \"groupName\": \"bottom\",\n" +
+                "    \"parent\": \"2\"\n" +
+                "  }\n" +
+                "]";
+        outputStream.write(content.getBytes());
+        Thread.sleep(1000);
+
+        // Action
+        new GroupDAO(testGroups);
```

Comment:
You are missing assertions here...

Reply:
yup this is meant to test that the action doesn't throw an exception
(I think it makes sense to have this if you compare it to the other tests in the class)